### DISCUSSION TITLE
Optional prefix

### DISCUSSION THREAD

**Discussion by dorshinar** [dcastil/tailwind-merge#296] [Ideas] [answered]
Hi :) 
I'd like to ask for a way to set optional prefixes so that they will be collectively merged, like so:

```js
twMerge('tw-text-slate-100 yo-text-slate-100 text-slate-200') === 'text-slate-200'
```

We have a component library that is prefixed to avoid clashes with applications consuming it, which in turn may or may not have a prefix defined. 
So let's say our `<Button />` component uses a `tw-` prefix, but the application that consumes it uses `ap-`. We'd like our `<Button />` to be conposable in a way that allows passing classNames that override the built-in one where it makes sense.

**Comment by dcastil** [comment, marked as answer]
Hey @dorshinar! 👋

You can do that already! It's not very pretty, but it does what you want:

```ts
const twMerge = extendTailwindMerge((config) => {
    Object.keys(config.classGroups).forEach((key) => {
        const value = config.classGroups[key]
        config.classGroups[key] = [...value, { tw: value, yo: value }]
    })

    return config
})

twMerge('tw-text-slate-100 yo-text-slate-100 text-slate-200')
// → 'text-slate-200'
```

But keep in mind that every prefix you use makes the internal data structure tailwind-merge computes exponentially bigger and might make the first call to `twMerge` slower (subsequent calls don't get slower). You can see in this line

```ts
config.classGroups[key] = [...value, { tw: value, yo: value }]
```

that we're making each class group essentially three times bigger. I think for one or two prefixes that doesn't matter, but if you're thinking about more, I'd check the tailwind-merge performance for that.

---

Related: https://github.com/dcastil/tailwind-merge/discussions/162

**Reply by dorshinar** [reply]
Thanks for the quick reply!
I'll give it a shot and update here

**Reply by sirajtahra** [reply]
hi @dcastil - this is no longer working with tailwind v4 prefixes, any idea ?

```
twMerge("tw:px-4 px-2")

// logs:  tw:px-4 px-2
```

**Reply by dcastil** [reply]
@sirajtahra use the approach explained here: https://github.com/dcastil/tailwind-merge/discussions/537#discussioncomment-12287367
